currency_rate_update/models/res_currency_rate_provider.py

```diff
@@ -0,0 +1,313 @@
+    )
+    currency_ids = fields.Many2many(
+        string='Currencies',
+        comodel_name='res.currency',
+        column1='provider_id',
+        column2='currency_id',
+        help='Currencies to be updated by this provider',
+        domain=lambda self: self._domain_currency_ids(),
+        track_visibility='onchange',
+    )
+    name = fields.Char(
+        string='Name',
+        compute='_compute_name',
+    )
+    interval_type = fields.Selection(
+        string='Units of scheduled update interval',
+        selection=[
+            ('days', 'Day(s)'),
+            ('weeks', 'Week(s)'),
+            ('months', 'Month(s)'),
+        ],
+        default='days',
+        required=True,
+        track_visibility='onchange',
+    )
+    interval_number = fields.Integer(
+        string='Scheduled update interval',
+        default=1,
+        required=True,
+        track_visibility='onchange',
+    )
+    update_schedule = fields.Char(
+        string='Update Schedule',
+        compute='_compute_update_schedule',
+    )
+    next_run = fields.Date(
+        string='Next scheduled update',
+        default=fields.Date.today(),
+        required=True,
+    )
+
+    _sql_constraints = [
+        (
+            'service_company_id_uniq',
+            'UNIQUE(service, company_id)',
+            'Service can only be used in one provider per company!'
+        ),
+        (
+            'valid_interval_number',
+            'CHECK(interval_number > 0)',
+            'Scheduled update interval must be greater than zero!'
+        )
+    ]
+
+    @api.onchange('service')
+    def _onchange_service(self):
+        res = {
+            'domain': {
+                'currency_ids': self._domain_currency_ids(),
+            },
+        }
+
+        return res
+
+    @api.onchange('currency_ids')
+    def _onchange_currency_ids(self):
+        res = {
+            'domain': {
+                'currency_ids': self._domain_currency_ids(),
+            },
+        }
+
+        return res
+
+    @api.multi
+    @api.depends('service')
+    def _compute_name(self):
+        for provider in self:
+            provider.name = list(filter(
+                lambda x: x[0] == provider.service,
+                self._selection_service()
+            ))[0][1]
+
+    @api.multi
+    @api.depends('active', 'interval_type', 'interval_number')
+    def _compute_update_schedule(self):
+        for provider in self:
+            if not provider.active:
+                provider.update_schedule = _('Inactive')
+                continue
+
+            provider.update_schedule = _('%(number)s %(type)s') % {
+                'number': self.interval_number,
+                'type': list(filter(
+                    lambda x: x[0] == provider.interval_type,
+                    self._fields['interval_type'].selection
+                ))[0][1],
+            }
+
+    @api.multi
+    def _domain_currency_ids(self):
+        domain = [('active', '=', True)]
```

Comment:
This domain is redundant

Reply:
Why?

Reply:
Because it's the domain that already applies ORM if you don't pass `active_test=False` in context.

Reply:
![giphy](https://user-images.githubusercontent.com/7165771/54241698-27c5f480-4522-11e9-959c-6d3b9912778e.gif)